FXyz-Importers/src/main/java/org/fxyz3d/importers/obj/ObjImporter.java

```diff
@@ -134,22 +134,22 @@ private ObjModel read(URL url, boolean asPolygon) {
 
     private static class ObjModel extends Model3D {
 
-        private static final Map<Predicate<String>, BiConsumer<String, ObjModel>> PARSERS = Map.of(
-                l -> l.startsWith("g ") || l.equals("g"), (l, m) -> m.parseGroupName(l),
-                l -> l.startsWith("v "),                  (l, m) -> m.parseVertex(l),
-                l -> l.startsWith("vt "),                 (l, m) -> m.parseVertexTexture(l),
-                l -> l.startsWith("f "),                  (l, m) -> m.parseFace(l),
-                l -> l.startsWith("s "),                  (l, m) -> m.parseSmoothGroup(l),
-                l -> l.startsWith("mtllib "),             (l, m) -> m.parseMaterialLib(l),
-                l -> l.startsWith("usemtl "),             (l, m) -> m.parseUseMaterial(l),
-                l -> l.startsWith("vn "),                 (l, m) -> m.parseVertexNormal(l),
-                // comments and empty lines are ignored
-                l -> l.isEmpty() || l.startsWith("#"),    (l, m) -> {});
+        // obj format spec: http://paulbourke.net/dataformats/obj/
+        private static final Map<String, BiConsumer<String, ObjModel>> PARSERS = Map.of(
+                "g",       (l, m) -> m.parseGroupName(l),
```

Comment:
`l.startsWith("g ") || l.equals("g")` is now converted to `l.startsWith("g")`, which is not exactly the same. 
Is this refactoring really necessary? I know you don't like the use of `substring` ...

Reply:
The parser method was changed to take care of the group name. `g` is the identifier, and if there's something after it or not is not something the check condition should care about, but the job of the parser.

Refactoring is not necessary, but it helps. If the importer is improved with more parsing options it makes it a bit easier to do so.

Reply:
Ok, your approach works as the parser in this case is just taking everything that comes after `g`, hoping that there won't be other than these two cases:
`g`
`g <group name>`

In case some file contains something like:
`group name`
it will be parsed as "roup name", but either way that won't have any kind of repercusion.

Reply:
`group` is not a valid start of line token. In fact, after `g` there can be more than 1 name separated by spaces, like
`g name1 name2`, which the parser doesn't treat.

I will also mention that the obj importer is incomplete in many other respects, and the mtl importer also doesn't account for all possibilities, from what I saw in the specs. The refactoring and cleanings of these will just make it easier to bring them up to specs in the future.